src/NeoSharp.BinarySerialization/Interfaces/IBinarySerializer.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.IO;
+
+namespace NeoSharp.BinarySerialization.Interfaces
```

Comment:
My suggestion is to move out IBinarySerializer as it was before and leave this namespace for SerializationHooks as IPre/PostSerialization. Maybe even to rename this accordingly if I got the idea right.